### PR TITLE
Introduce data schema to store raw tensors

### PR DESCRIPTION
**Pull request by lucylq**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #6540

Differential Revision: [D65156641](https://our.internmc.facebook.com/intern/diff/D65156641)

### CODE REVIEW

**Comment by lucylq** [comment]
See: https://github.com/pytorch/executorch/issues/6492

File: schema/data.fbs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  //   "batch".
+  // - (0, 2, 1) represents a [row, batch, column] ordering where "column" is
+  //   the innermost dimension, then comes "batch", and the outermost dimension
+  //   is "row".
+  dim_order:[ubyte];
+
+  // Offset in scalar_type elements (e.g., multiples of 4 bytes for an int
+  // scalar type) from the beginning of the tensor buffer to the beginning of
+  // the actual data. Currently, the runtime only supports a value of zero.
+  storage_offset:int;
```
Comment (lucylq) [inline schema/data.fbs:32]: `storage_offset` and `layout` taken from `program.fbs`, don't think there is an immediate use case for them.

File: schema/data.fbs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+file_identifier "DT01";
+file_extension "data";
+
+table TensorMetadata {
+  // The unique id used to connect the data and program.
+  fully_qualified_name:string;
+  scalar_type:ScalarType;
+
+  // Size of each dimension.
+  dim_sizes:[int];
```
Comment (lucylq) [inline schema/data.fbs:14]: equivalent of 'sizes' in `program.fbs`, using `dim_sizes` to differentiate from `size` of the tensor below.

**Comment by lucylq** [comment]
@lucylq has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D65156641).

**Comment by lucylq** [comment]
@lucylq has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D65156641).

File: exir/schema_data.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+from executorch.exir.scalar_type import ScalarType
+
+# Note: check executorch/schema/data.fbs for explanations of these fields.
+
+
+@dataclass
+class TensorMetadata:
+    fully_qualified_name: str
+    scalar_type: ScalarType
+    dim_sizes: List[int]
```
Comment (JacobSzwejbka) [inline exir/schema_data.py:20]: just sizes

**Review by JacobSzwejbka — APPROVED** [review]
I was thinking maybe we should try to reuse more fields from the core schema but I guess the goal of this is different enough that we dont have to. Are we 100% sure we even need sizes and dim order. I think the set of fully qualified names provides a good enough compatibility guarantee (seems unlikely 2 models would have the exact same key set but different values)

**Review by dbort — APPROVED** [review]

**Pull request merged**